docs/source/array-creation.rst

```diff
@@ -304,3 +304,74 @@ destinations:
 .. code-block:: Python
 
    >>> da.store([array1, array2], [output1, output2])  # doctest: +SKIP
+
+
+Plugins
+=======
+
+We can run arbitrary user-defined functions on dask.arrays whenever they are
+constructed. This allows us to build a variety of custom behaviors that improve
+debugging, user warning, etc..  You can register a list of functions to run on
+all dask.arrays to the global ``array_plugins=`` value:
+
+.. code-block:: python
+
+   >>> def f(x):
+   ...     print(x.nbytes)
+
+   >>> with dask.set_options(array_plugins=[f]):
+   ...     x = da.ones((10, 1), chunks=(5, 1))
+   ...     y = x.dot(x.T)
+   80
+   80
+   800
+   800
+
+If the plugin function returns None then the input Dask.array will be returned
+without change.  If the plugin function returns something else then that value
+will be the result of the constructor.
+
+Examples
+--------
+
+Automatically compute
+~~~~~~~~~~~~~~~~~~~~~
+
+We may wish to turn some Dask.array code into normal NumPy code.  This is
+useful for example to track down errors immediately that would otherwise be
+hidden by Dask's lazy semantics.
+
+.. code-block:: python
+
+   >>> with dask.set_options(array_plugins=[lambda x: x.compute()]):
```

Comment:
Note that in practice, a plugin `lambda x: x.persist()` or `lambda x: da.asarray(x.compute())` might work better useful, since the it keeps the result as dask arrays.